src/imports.js

```diff
@@ -132,12 +135,20 @@ function makeSortedItems(items, outerGroups) {
 
 // Exclude "ImportDefaultSpecifier" – the "def" in `import def, {a, b}`.
 function getSpecifiers(importNode) {
-  return importNode.specifiers.filter((node) => isImportSpecifier(node));
+  switch (importNode.type) {
+    case "TSImportEqualsDeclaration":
+      return [];
+    default:
+      return importNode.specifiers.filter((node) => isImportSpecifier(node));
+  }
 }
 
 // Full import statement.
 function isImport(node) {
-  return node.type === "ImportDeclaration";
+  return (
+    node.type === "ImportDeclaration" ||
+    node.type === "TSImportEqualsDeclaration"
```

Comment:
As far as I can tell, `import A = B` and `export import A = B` parse into the exact same AST, except that the latter has `isExport: true`.

It seems like the current code will treat `export import A = B` exactly like `import A = B`.

I have no clue what `export import A = B` even does. Do you?

Is it weird that something that looks like an export can be sorted among imports, unlike all other types of exports? I guess it should be supported in the exports rule?

Reply:
> Do you?

I know [only one]([re-export Typescript enum from namespace?](https://stackoverflow.com/questions/47701270/re-export-typescript-enum-from-namespace)) usage of `export import`.

Reply:
> I guess it should be supported in the exports rule?

I don't know. `export import` it is export and import at the same time. Maybe add an option to configure it? User should can decide how sorting it.

Reply:
I suggest thinking about this in the next pr, but for now just exclude it.

Reply:
> export import it is export and import at the same time.

There’s also `export { thing } from "module"` which is also import and export at the same time? So what’s the difference with `export import A = B`?

> Maybe add an option to configure it? User should can decide how sorting it.

Why? What is there to configure?

> I suggest thinking about this in the next pr, but for now just exclude it.

I feel a bit uneasy about merging something incomplete or not fully thought through.

Reply:
> So what’s the difference with `export import A = B`?

`export { thing } from "module"` is not an import, because you can't use `thing` after this line. But you can use `A` after `export import A = B`.

Reply:
> I feel a bit uneasy about merging something incomplete or not fully thought through.

OK. Let's come to a conclusion to which category they belong.

Reply:
> But you can use A after export import A = B.

Oh wow. That’s weird. 🙈 Maybe that’s why it’s the same AST node then. And maybe this is a good motivation to categorize them as imports – if so reverting https://github.com/lydell/eslint-plugin-simple-import-sort/pull/149/commits/3de145b4595f145f4c8426c50d5165fd0b2893d0 should be good?